var/spack/repos/builtin/packages/gatk/package.py

```diff
@@ -3,59 +3,91 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-from spack import *
 import os.path
+import glob
+from spack import *
 
 
 class Gatk(Package):
-    """Genome Analysis Toolkit
-       Variant Discovery in High-Throughput Sequencing Data
     """
+    Genome Analysis Toolkit
+    Variant Discovery in High-Throughput Sequencing Data
+    """
+
     homepage = "https://software.broadinstitute.org/gatk/"
-    url      = "https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip"
-
-    version('4.0.11.0', sha256='5ee23159be7c65051335ac155444c6a49c4d8e3515d4227646c0686819934536')
-    version('4.0.8.1', sha256='6d47463dfd8c16ffae82fd29e4e73503e5b7cd0fcc6fea2ed50ee3760dd9acd9',
-            url='https://github.com/broadinstitute/gatk/archive/4.0.8.1.tar.gz')
-    version('4.0.4.0', '083d655883fb251e837eb2458141fc2b',
-            url="https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip")
-    version('3.8-1', 'a0829534d2d0ca3ebfbd3b524a9b50427ff238e0db400d6e9e479242d98cbe5c', extension='tar.bz2',
-            url="https://software.broadinstitute.org/gatk/download/auth?package=GATK-archive&version=3.8-1-0-gf15c1c3ef")
-    version('3.8-0', '0581308d2a25f10d11d3dfd0d6e4d28e', extension='tar.gz',
-            url="https://software.broadinstitute.org/gatk/download/auth?package=GATK")
-
-    depends_on('java@8:', type='run')
-    depends_on('python@2.6:2.8,3.6:', type='run', when='@4.0:')
-    depends_on('r@3.2:', type='run', when='@4.0:')
+    url = "https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip"
+
+    version(
+        "4.0.12.0",
+        sha256="733134303f4961dec589247ff006612b7a94171fab8913c5d44c836aa086865f",
+        url="https://github.com/broadinstitute/gatk/releases/download/4.0.12.0/gatk-4.0.12.0.zip",
+        preferred=True,
+    )
+    version(
+        "4.0.11.0",
+        sha256="5ee23159be7c65051335ac155444c6a49c4d8e3515d4227646c0686819934536",
+        url="https://github.com/broadinstitute/gatk/releases/download/4.0.11.0/gatk-4.0.11.0.zip",
+    )
+    version(
+        "4.0.8.1",
+        sha256="6d47463dfd8c16ffae82fd29e4e73503e5b7cd0fcc6fea2ed50ee3760dd9acd9",
+        url="https://github.com/broadinstitute/gatk/archive/4.0.8.1.tar.gz",
```

Comment:
This link for version 4.0.8.1 is wrong -- the correct link is `https://github.com/broadinstitute/gatk/releases/download/4.0.8.1/gatk-4.0.8.1.zip`

What you've linked to here is actually the source tarball rather than the compiled release.

Reply:
Added 4.1.0.0 and fixed the url for 4.0.8.1.